kong/init.lua

```diff
@@ -769,6 +731,10 @@ function Kong.access()
     ctx.workspace = old_ws
   end
 
+  if plugins_iterator.has_response then
```

Comment:
As noted below, I feel this is leaking across the requests, so I would like to know more that it is not so.